src/config.py

```diff
@@ -19,21 +18,39 @@
 # build directory. Debug symbols and __FILE__ use absolute paths.
 DIR_BUILDROOT = '/usr/obj/cloudabi-ports'
 
-# Location where resource files are stored.
-DIR_RESOURCES = os.path.join(os.getcwd(), 'misc')
+
+RESOURCES = dict((name, get_data('misc', name))
+                 for name in ['config.guess', 'config.sub'])
 
 # Location at which distfiles can be fetched in case the master sites
 # are down.
 FALLBACK_MIRRORS = {'https://nuxi.nl/distfiles/third_party/'}
 
-# Host C and C++ compiler, used to compile the build tools. We'd better
-# use Clang if available. Compared to GCC, it has the advantage that it
-# does not depend on the 'as' and 'ld' utilities being part of $PATH.
-HOST_CC = ('/usr/bin/clang-3.7' if platform.system() == 'Linux' else
-           '/usr/bin/cc')
-HOST_CXX = ('/usr/bin/clang++-3.7' if platform.system() == 'Linux' else
-            '/usr/bin/c++')
-
-# Name of the Perl executable.
-PERL = ('/usr/local/bin/perl' if platform.system() == 'FreeBSD' else
-        '/usr/bin/perl')
+
+class Platform(object):
```

Comment:
If we're going in this direction, would it make sense to use some kind of class hierarchy here, so we'll have a `FreeBSDPlatform` that inherits from this, but overrides `perl()`?